pkg/operator/staticresourcecontroller/static_resource_controller.go

```diff
@@ -240,6 +248,53 @@ func (c *StaticResourceController) Name() string {
 	return "StaticResourceController"
 }
 
+func (c *StaticResourceController) RelatedObjects() ([]configv1.ObjectReference, error) {
+	if c.restMapper == nil {
+		return nil, errors.New("StaticResourceController.restMapper is nil")
+	}
+
+	acc := make([]configv1.ObjectReference, 0)
+	errors := []error{}
+
+	for _, file := range c.files {
```

Comment:
I'd like to be smarter.  I think the rules are closer to 
1. if cluster scoped resource, list that exact resource
2. if namespace scoped and included in the "all" alias, just include the namespace.  de-dupe namespaces
3. if namespace and not included in the "all" alias, list that exact resource.

Reply:
See https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/restmapper/category_expansion.go#L48-L88 for how to determine "all"

Reply:
👍🏽

Reply:
I think it's not necessary to add namespaces to RelatedObjects - someone must have created the namespace and therefore add it it its RelatedObjects.

(At least in storage operators we have them hardcoded in RelatedObjects).

Reply:
+1

Reply:
> I think it's not necessary to add namespaces to RelatedObjects - someone must have created the namespace and therefore add it it its RelatedObjects.
> 
> (At least in storage operators we have them hardcoded in RelatedObjects).

The idea is avoid an explosion of individual resources that are not required and slow down collection.  Right now you're iterating through manifests and enumerating all of them.  This will result in entries like
1. ns=foo secret=one
2. ns=foo configmap=two
3. ns=foo serviceaccount=three
4. ns=foo
5. ns=foo rolebinding=four
6. ns=foo servicemonitor=five

What we want is to collapse all that to
1. ns=foo
2. ns=foo servicemonitor=five

Because we know that oc adm inspect will collect the "all" category for ns=foo and collect the individual resources for "all".  And servicemonitor is not in "all" (I don't think).

if you don't inspect all, please at least decimate resources by namespace for 4.8.

Reply:
> What we want is to collapse all that to
> 1. ns=foo
> 2. ns=foo servicemonitor=five

I am arguing that we can skip "1. ns=foo", unless the `kind: namespace` is directly in the controller assets.

At least in storage operators we either create namespaces with StaticResourceController, and hence it will be returned in `RelatedObjects()` here as any other non-namespaced object , or CVO creates it from `manifests/` and we have it hardcoded in RelatedObjects in the operator.

IMO StaticResourceController should not report objects it does not create. Something else must have created them and that component is responsible for reporting them.

Reply:
> IMO StaticResourceController should not report objects it does not create. Something else must have created them and that component is responsible for reporting them.

I can accept that position as a starting point.  In my example, the namespace was created as item 4.